cmake: searching across directories for config files

This can be recreated on at least macOS, due to #32707. 

If there are two checkouts of the source, the build in one directory, can fail to configure, because of files in the other directory:
```bash
git clone https://github.com/bitcoin/bitcoin/ bitcoin_a
git clone https://github.com/bitcoin/bitcoin/ bitcoin_b

cd bitcoin_a
make -C depends HOST=x86_64-w64-mingw32
# fails to build libevent (#32707)
cd ../bitcoin_b
# fails to configure, because of the libevent config files (for a different target) in the other checkout
cmake -B build
<snip>
-- Performing Test NO_DIAGNOSTICS_BOOST_NO_CXX98_FUNCTION_BASE
-- Performing Test NO_DIAGNOSTICS_BOOST_NO_CXX98_FUNCTION_BASE - Failed
CMake Error at /Users/xxx/bitcoin_a/depends/work/build/x86_64-w64-mingw32/libevent/2.1.12-stable-f0fd46dc234/build/LibeventConfig.cmake:75 (message):
  Your libevent library does not contain a pthreads component!

  The valid components are core, extra, pthreads and openssl.
Call Stack (most recent call first):
  /Users/xxx/bitcoin_a/depends/work/build/x86_64-w64-mingw32/libevent/2.1.12-stable-f0fd46dc234/build/LibeventConfig.cmake:90 (no_component_msg)
  cmake/module/FindLibevent.cmake:54 (find_package)
  CMakeLists.txt:416 (find_package)
```

--- a/depends/funcs.mk
+++ b/depends/funcs.mk
@@ -224,6 +224,7 @@ $(1)_cmake=env CC="$$($(1)_cc)" \
                -DCMAKE_INSTALL_LIBDIR=lib/ \
                -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
                -DCMAKE_VERBOSE_MAKEFILE:BOOL=$(V) \
+               -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY:BOOL=TRUE \
                $$($(1)_config_opts)
 ifeq ($($(1)_type),build)
 $(1)_cmake += -DCMAKE_INSTALL_RPATH:PATH="$$($($(1)_type)_prefix)/lib"

